File: 2021/09/24/PivotIndex.java
/**
 * https://leetcode.com/problems/find-pivot-index/
 * @author Andrew Jarombek
 * @since 9/24/2021
 */

public class PivotIndex {
    public int pivotIndex(int[] nums) {
        int left = 0;
        int right = 0;

        for (int num: nums) {
            left += num;
        }

        for (int i = 0; i < nums.length; i++) {
            left -= nums[i];

            if (left == right) {
                return i;
            }

            right += nums[i];
        }

        return -1;
    }
}

File: 2022/07/22/pivotIndex.js
/**
 * https://leetcode.com/problems/find-pivot-index/
 * @author Andrew Jarombek
 * @since 7/22/2022
 */

const pivotIndex = function(nums) {
    let left = nums.reduce((acc, val) => acc + val);
    let right = 0;

    for (let i = 0; i < nums.length; i++) {
        const val = nums[i];
        left -= val;

        if (right === left)
            return i;

        right += val;
    }

    return -1;
}

let result = pivotIndex([2, 1, -1]);
console.assert(result === 0);

result = pivotIndex([-1, 1, 2]);
console.assert(result === 2);